src/app/song/songs.service.ts

```diff
@@ -65,14 +69,34 @@ export class SongService{
     }
 
     deleteSong(songId: string){
+        console.log("deleted");
         this.Http.delete('http://localhost:3000/api/songs/' + songId)
         .subscribe(() => {
-            const updatedSongs = this.songs.filter(song => songId !== songId);
+            const updatedSongs = this.songs.filter(song => song.id !== songId);
             this.songs = updatedSongs;
             this.songsUpdated.next([...this.songs]);
+
             this.notificationService.submitMessage("Song was deleted successfully");
         });
     }
+
+    updateSong(id: string, name: string, genre: Genre, song_path: string, image_path: string, release_date: Date,
+        artists: string[], //TODO: change to artist array
+        num_of_times_liked: number){
+            const song: Song = {
+                id: id, 
+                name: name, 
+                genre: genre, 
+                song_path: song_path, 
+                image_path: image_path, 
+                release_date: release_date,
+                artists: artists, 
+                num_of_times_liked: num_of_times_liked
+            };
+            this.Http.put("http://localhost:3000/api/songs/" + id, song).subscribe(res => {
```

Comment:
Same as above, do something like endpoint="http://localhost:3000"
and then here:
Http.put(endpoint + "/api/songs")